### ISSUE TITLE
Kernel fix for compressed data corruption found

### ISSUE THREAD

**Issue by Zygo** [Zygo/bees#107]
bees hits this kernel bug quite often.  Now there's a fix!

Reference:  https://www.mail-archive.com/linux-btrfs@vger.kernel.org/msg85492.html

The entire kernel patch is:

```
diff --git a/fs/btrfs/extent_io.c b/fs/btrfs/extent_io.c
index 52abe4082680..d7b31991785a 100644
--- a/fs/btrfs/extent_io.c
+++ b/fs/btrfs/extent_io.c
@@ -2985,7 +2985,7 @@ static int __do_readpage(struct extent_io_tree *tree,
                 */
                if (test_bit(EXTENT_FLAG_COMPRESSED, &em->flags) &&
                    prev_em_start && *prev_em_start != (u64)-1 &&
-                   *prev_em_start != em->orig_start)
+                   *prev_em_start != em->start)
                        force_bio_submit = true;
 
                if (prev_em_start)


```

It Works For Me(tm).

Various doc updates will be required for bees as the fix finds its way to mainline kernels.

**Comment by Zygo** [comment]
The patch has doubled in size, now it's two lines:

https://www.mail-archive.com/linux-btrfs@vger.kernel.org/msg85514.html

**Comment by speed47** [comment]
FWIW, this patch has been merged during the kernel 5.1 merge window. We're approaching the end of the release candidates dance, v5.1 should be released in a few days.

**Comment by Zygo** [comment]
The fix has also been backported to all of the active stable branches (5.0, 4.19, 4.14, 4.9, 4.4, even 3.x).
